Sources/HipMobileUI/ViewModels/Pages/AppetizerPageViewModel.cs

```diff
@@ -104,17 +109,23 @@ public AppetizerPageViewModel(Exhibit exhibit)
         /// Sets the average user rating, the count and the stars if the user is connected to the internet.
         /// </summary>
         /// <returns></returns>
-        private async void SetUserRating()
+        private async void RefreshUserRating()
         {
             if (IoCManager.Resolve<INetworkAccessChecker>().GetNetworkAccessStatus() == NetworkAccessStatus.NoAccess)
             {
                 RatingAverage = "-";
                 RatingStars = "?????";
                 RatingCount = "- " + Strings.UserRating_Rate_Count;
+                UserDialogs.Instance.Alert(new AlertConfig()
+                {
+                    Title = Strings.UserRating_Dialog_Title_No_Internet,
+                    Message = Strings.UserRating_Dialog_Message_No_Internet,
+                    OkText = Strings.UserRating_Ok
+                });
             }
             else
             {
-                var userRating = await IoCManager.Resolve<IUserRatingManager>().GetUserRatingAsync(exhibit);
+                var userRating = await IoCManager.Resolve<IUserRatingManager>().GetUserRatingAsync(exhibit.IdForRestApi);
```

Comment:
This can throw in case the network is down, right? The check above with the `INetworkAccessChecker` doesn't ensure that the network will not go down shortly afterwards. You should wrap this in a try-catch with error handling, otherwise the app might crash on network errors.